keras/backend/tensorflow_backend.py

```diff
@@ -2486,13 +2486,33 @@ def __init__(self, inputs, outputs,
             raise ValueError('Some keys in session_kwargs are not '
                              'supported at this '
                              'time: %s', session_kwargs.keys())
+        self._extra_fetches_info = dict()
         self._callable_fn = None
-        self._merged_summaries = None
         self._feed_arrays = None
         self._feed_symbols = None
         self._symbol_vals = None
         self._session = None
 
+    def _add_fetch(self, fetch, fetch_callback):
+        fetch_index = len(self.fetches)
+        self.fetches.append(fetch)
+        self._extra_fetches_info[fetch_index] = fetch_callback
+        if self._callable_fn is not None:
+            self._make_callable(self._feed_arrays,
+                                self._feed_symbols,
+                                self._symbol_vals,
+                                get_session())
```

Comment:
I think it's easier to just set `self._callable_fn = None` here. The callable will be made on the next call to the function.